src/mutations/authentication.ts

```diff
@@ -79,4 +87,27 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
       notifier?.({ type: signOutRoutine.FAILURE, payload: { error } });
     },
   });
+
+  queryClient.setMutationDefaults(MUTATION_KEYS.SWITCH_MEMBER, {
+    mutationFn: async (args: { memberId: string; domain: string }) => {
+      // get token from stored sessions given memberId
+      const sessions = getStoredSessions();
+      const session = sessions?.find(
+        ({ id: thisId }) => args.memberId === thisId,
+      );
+      if (!session) {
+        throw new Error('Session is invalid');
+      }
+      setCurrentSession(session.token, args.domain);
+      return args.memberId;
+    },
+    onSuccess: () => {
+      notifier?.({ type: switchMemberRoutine.SUCCESS });
+      // reset queries to take into account the new token
+      queryClient.resetQueries();
+    },
+    onError: (error) => {
+      notifier?.({ type: switchMemberRoutine.FAILURE, payload: { error } });
```

Comment:
Nitpick: we are often checking that the notifier is well-defined. However it is my understanding that the notifier should be available once the app is loaded. Would it be possible to change the check and types such that it is only performed once (which would avoid reverifying the nullability of the object on every usage)? I believe `QueryConfig` is locally controllable? Or we could re-type